engine/gamesys/src/gamesys/resources/res_texture.cpp

```diff
@@ -386,8 +406,15 @@ namespace dmGameSystem
     dmResource::Result ResTextureDestroy(const dmResource::ResourceDestroyParams* params)
     {
         TextureResource* texture_res = (TextureResource*) dmResource::GetResource(params->m_Resource);
-        dmGraphics::DeleteTexture(texture_res->m_Texture);
-        delete texture_res;
+
+        if (texture_res->m_Uploading)
+        {
+            // The job is currently in flight
+            texture_res->m_DelayDelete = 1;
+            return dmResource::RESULT_OK;
+        }
```

Comment:
I wonder if we want to move more resource payloads to be "async-loadable", e.g any *other* graphics API allows you to upload data in a separate thread (OpenGL only allows a few resource types I Think), so we probably want to keep that in mind when designing render script 2.0 🤔 (just a thought)

Reply:
Yes, I thought the same. We probably want more resource types to be be possibly loaded at the same time.
Currently, the loader is single threaded, waiting for each resource to finish.